packages/ibm-products/src/components/SearchBar/SearchBar.test.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react'; // https://testing-library.com/docs/react-testing-library/intro
+
+import { pkg, carbon } from '../../settings';
+import uuidv4 from '../../global/js/utils/uuidv4';
+import { SearchBar } from '.';
+
+const blockClass = `${pkg.prefix}--search-bar`;
+const componentName = SearchBar.displayName;
+const dataTestId = uuidv4();
+const mockOnSubmit = jest.fn();
+const mockOnChange = jest.fn();
+const value = 'Carbon';
+
+const defaultProps = {
+  clearButtonLabelText: 'Clear',
+  placeHolderText: 'Search...',
+  submitLabel: 'Search',
+  onChange: mockOnChange,
+  onSubmit: mockOnSubmit,
+};
+
+const scopes = [
+  {
+    id: '2',
+    text: 'Scope 2',
+  },
+  {
+    id: '1',
+    text: 'Scope 1',
+  },
+  {
+    id: '3',
+    text: 'Scope 3',
+  },
+];
+
+const scopesDefaultProps = {
+  ...defaultProps,
+  scopes,
+  scopesTypeLabel: 'Scopes',
+};
+
+const renderComponent = (props = defaultProps) => {
+  return render(<SearchBar {...props} />);
+};
+
+describe(componentName, () => {
+  it('renders with default props', async () => {
+    renderComponent();
+
+    const searchBox = screen.getByRole('searchbox');
+    const submitButton = screen.getByText(defaultProps.submitLabel);
+
+    expect(searchBox).toBeInTheDocument();
+    expect(searchBox.value).toBe('');
+    expect(searchBox.placeholder).toBe(defaultProps.placeHolderText);
+    expect(submitButton).toHaveTextContent(defaultProps.submitLabel);
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toBeDisabled();
+  });
+
+  it('has initial value, it enables submit button, and click clear button to clear input field', async () => {
+    renderComponent({ ...defaultProps, value });
+
+    const searchBox = screen.getByDisplayValue(value);
+    const submitButton = screen.getByText(defaultProps.submitLabel);
+    const clearButton = screen.getByTitle(defaultProps.clearButtonLabelText);
+
+    expect(clearButton).toBeInTheDocument();
+    expect(searchBox).toBeInTheDocument();
+    expect(submitButton).toBeEnabled();
+
+    fireEvent.click(submitButton);
```

Comment:
We've been trying to only use `userEvent` for interactions, ie:
```js
const { click } = userEvent;
await click(submitButton);
```

Reply:
Resolved.